pkg/patterns/addon/pkg/loaders/types.go

```diff
@@ -132,21 +132,25 @@ func (r *FSRepository) LoadManifest(ctx context.Context, packageName string, id
 }
 
 type Channel struct {
-	Manifests []Version `json:"manifests,omitempty"`
+	Packages []Version `json:"packages,omitempty"`
 }
 
 type Version struct {
-	Version string
+	Package string `json:"name"`
+	Version string `json:"version"`
 }
 
-func (c *Channel) Latest() (*Version, error) {
+func (c *Channel) Latest(packageName string) (*Version, error) {
 	var latest *Version
-	for i := range c.Manifests {
-		v := &c.Manifests[i]
+	for i := range c.Packages {
+		v := &c.Packages[i]
+		if v.Package != "" && v.Package != packageName {
+			continue
+		}
 		if latest == nil {
 			latest = v
-		} else {
-			return nil, fmt.Errorf("version selection not implemented")
+		} else if latest.Compare(v) < 0 {
+			latest = v
 		}
```

Comment:
Isn't `else {...}` needed?
If both `if{...}` and `else if ... {...}` aren't match its state, this function returns `latest` as nil. Or is it ok?

Reply:
The missing if branch is when we already have a newer version of the package.

e.g.

```
packages:
- name: foo
  version: 2
- name: foo
  version: 1
```

latest will be set to the the foo v2 on the first iteration, and then we ignore foo v1 because it is older.

I'm not sure how useful it is to actually have multiple matching packages, but the scenario I wanted to enable was this:

```
packages:
- version: 2.0.0 # default version
- name: foo
  version: 1.0.0 # foo is pinned to a particular version
```

Reply:
I see. :smiley: 
I recognize that I have misunderstanded a logic..
Thanks to your kind explanation.